packages/form-builder/src/hooks/useFieldNames.js

```diff
@@ -0,0 +1,39 @@
+import {useSelect} from "@wordpress/data";
+
+export const getFieldNameFrequency = (fieldName, fieldNames) => {
+    return fieldNames.filter(name => name === fieldName).length
+}
+
+export const getFieldNameSuggestion = (name, names) => {
+    const [ prefix ] = name.split(/^(.*)-([0-9]*)$/g).filter(Boolean)
+    const similarFieldNames = names.filter(fieldName => fieldName.startsWith(prefix));
+    const increments = similarFieldNames.flatMap(fieldName => fieldName.split(/^.*-([0-9]*)$/g).filter(Number) ) || [ 0 ]
+    const nextIncrement = increments.length ? Math.max(...increments) + 1 : 1
+    return `${prefix}-${nextIncrement}`;
+}
+
+export const flattenBlocks = (block) => [block, ...block.innerBlocks.flatMap(flattenBlocks)]
```

Comment:
This looks nifty, can you please write some detailed descriptions of everything going on in this file for our future selves?